lib/takeaway.rb

```diff
@@ -0,0 +1,40 @@
+require_relative 'order'
+require_relative 'twilio_txt'
+
+class Takeaway
+  attr_reader :number, :order
+
+  def initialize(num = 'no_num',
+                 order_klass = Order.new,
+                 txt_klass = Sms.new)
+    @number = num
+    @order = order_klass
+    @txt = txt_klass
+  end
+
+
+  def add(food, quantity=1)
+    raise ArgumentError, "try again" if !@order.menu.has_key?(food.to_sym)
```

Comment:
Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`. (https://github.com/bbatsov/ruby-style-guide#hash-key)<br>Style/NegatedIf: Favor `unless` over `if` for negative conditions. (https://github.com/bbatsov/ruby-style-guide#unless-for-negatives)<br>Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)